packages/components/src/components/user/SiteUsersGridPanel.tsx

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2019 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
+ * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
+ */
+import * as React from 'react'
+import { List, Map } from "immutable";
+import { Button, Row, Col, MenuItem } from "react-bootstrap";
+import { Filter, ActionURL } from "@labkey/api";
+import { CreateUsersModal } from "./CreateUsersModal";
+import { QueryGridModel } from "../base/models/model";
+import { SCHEMAS } from "../base/models/schemas";
+import { getStateQueryGridModel } from "../../models";
+import { getQueryGridModel } from "../../global";
+import { ManageDropdownButton } from "../buttons/ManageDropdownButton";
+import { SelectionMenuItem } from "../menus/SelectionMenuItem";
+import { QueryGridPanel } from "../QueryGridPanel";
+import { UserDetailsPanel } from "./UserDetailsPanel";
+import { SecurityPolicy, SecurityRole } from "../permissions/models";
+import { UserActivateChangeConfirmModal } from "./UserActivateChangeConfirmModal";
+import { capitalizeFirstChar } from "../../util/utils"
+import { getLocation, getRouteFromLocationHash, replaceParameter } from "../../util/URL";
+import { getBrowserHistory } from "../../util/global";
+import { UserDeleteConfirmModal } from "./UserDeleteConfirmModal";
+
+const OMITTED_COLUMNS = List(['phone', 'im', 'mobile', 'pager', 'groups', 'active', 'hasPassword', 'firstName', 'lastName', 'description', 'expirationDate']);
+
+interface Props {
+    onCreateComplete: (response: any, role: string) => any
+    onUsersStateChangeComplete: (response: any) => any
+
+    policy: SecurityPolicy
+    rolesByUniqueName: Map<string, SecurityRole>
+
+    // optional array of role options, objects with id and label values
+    // note that the createNewUser action will not use this value but it will be passed back to the onCreateComplete
+    newUserRoleOptions?: Array<any>
+
+    // option to disable the delete UI pieces for this component
+    allowDelete?: boolean
+}
+
+interface State {
+    usersView: string
+    showDialog: string // valid options are 'create', 'deactivate', 'reactivate', 'delete', undefined
+    selectedUserId: number
+    unlisten: any
+}
+
+export class SiteUsersGridPanel extends React.PureComponent<Props, State> {
+
+    static defaultProps = {
+        allowDelete: true
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        const unlisten = getBrowserHistory().listen((location, action) => {
```

Comment:
Can you add a comment about why this listen is necessary here?

Reply:
Yes